operator/internal/manifests/var.go

```diff
@@ -273,3 +279,23 @@ func lokiReadinessProbe() *corev1.Probe {
 		FailureThreshold:    3,
 	}
 }
+
+func ensureCredentialsForGCS(podSpec *corev1.PodSpec, secretName string) {
+	podSpec.Volumes = append(podSpec.Volumes, corev1.Volume{
+		Name: secretName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: secretName,
+			},
+		},
+	})
+	podSpec.Containers[0].VolumeMounts = append(podSpec.Containers[0].VolumeMounts, corev1.VolumeMount{
+		Name:      secretName,
+		ReadOnly:  false,
+		MountPath: secretDirectory,
+	})
+	podSpec.Containers[0].Env = append(podSpec.Containers[0].Env, corev1.EnvVar{
+		Name:  EnvGoogleApplicationCredentials,
+		Value: path.Join(secretDirectory, GCSFileName),
+	})
```

Comment:
I suggest to continue using `mergo.Merge` for mutating manifests as we do in other places, e.g. [service monitor PKI](https://github.com/grafana/loki/blob/main/operator/internal/manifests/service_monitor.go#L136).

In addition I suggest we move the whole function out of `var.go` to `storage/configure.go` file and follow-up the configure pattern as in `gateway_tenants.go`, i.e.
- First in `storage/configure.go` you place the `ensureCredentialsFor GCS`
- Second in `manifests/object_storage.go` you place the switch based on `opts.Stack.Storage.Secret.Type`
- Third you call the `configureDeploymentForStorageType` and `configureStatefulSetForStorageType` only in the Builder functions (e.g. `BuildQuerier` w/o nesting if blocks in the original constructor functions (e.g. `NewQuerierDeployment`)

This ensures that we have dedicated configure/mutate functions with proper error handling (thanks to `mergo.Merge` while keeping our manifest constructors are pure and minimal as possible.

e.g.
```go
func configureDeploymentForStorageType(d *appsv1.Deployment, t lokiv1beta1.ObjectStorageType) error {
	switch t {
	case lokiv1beta1.ObjectStorageSecretGCS:
		return storage.ConfigureDeployment(d)
	default:
		return nil
	}
}
func configureStatefulSetForStorageType(d *appsv1.Deployment, t lokiv1beta1.ObjectStorageType) error {
	switch t {
	case lokiv1beta1.ObjectStorageSecretGCS:
		return storage.ConfigureStatefulSet(d)
	default:
		return nil
	}
}
```